### PR TITLE
[onnx]averagepool

### PR DESCRIPTION
**Pull request by xiaxin1998**

### CODE REVIEW

**Comment by xiaxin1998** [comment]
@zhichao-li

File: pyzoo/test/zoo/pipeline/utils/test_utils_onnx.py [hunk 1]
```diff
@@ -41,10 +41,13 @@ def dump_pytorch_to_onnx(self, pytorch_model, input_shape_with_batch):
     def compare_with_pytorch(self, pytorch_model, input_shape_with_batch):
         onnx_path = self.dump_pytorch_to_onnx(pytorch_model, input_shape_with_batch)
         onnx_model = onnx.load(onnx_path)
+        print(onnx_path)
```
Comment (zhichao-li) [inline pyzoo/test/zoo/pipeline/utils/test_utils_onnx.py]: remove?

File: pyzoo/test/zoo/pipeline/onnx/test_model_loading.py [hunk 2]
```diff
@@ -38,3 +38,17 @@ def test_onnx_conv2d_2(self):
         )
         input_shape_with_batch = (1, 3, 224, 224)
         self.compare_with_pytorch(pytorch_model, input_shape_with_batch)
+
+    def test_onnx_averagepool2d(self):
```
Comment (zhichao-li) [inline pyzoo/test/zoo/pipeline/onnx/test_model_loading.py]: add unitest for `pads` and `countIncludePad` ?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ZeroPadding2D.scala [hunk 3]
```diff
@@ -42,21 +45,47 @@ import scala.reflect.ClassTag
...
+    inputShape: Shape = null,
+    mode: String = "constant",
+    value: Int = 0)(implicit ev: TensorNumeric[T]): ZeroPadding2D[T] = {
     val paddingArray = padding match {
       case null => throw new IllegalArgumentException("For ZeroPadding2D, " +
         "padding can not be null, please input int tuple of length 2")
       case _ => Array(padding._1, padding._1, padding._2, padding._2)
     }
-    new ZeroPadding2D[T](paddingArray, KerasUtils.toBigDLFormat(dimOrdering), inputShape)
+    new ZeroPadding2D[T](
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ZeroPadding2D.scala]: revert the changes for `ZeroPadding2D`

File: pyzoo/zoo/pipeline/api/onnx/mapper/pad.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        assert len(self.inputs) == 1, "Conv accept single input only"
+        rank = len(self.inputs[0].get_input_shape())
+        if (rank == 4):  # NCHW
+            mode = self.onnx_attr['mode']
+            value = int(self.onnx_attr['value'])
+            assert mode == "constant", "Pad only accept constant mode"
+            assert value == 0, "Pad only accept Zero value"
+            dim_ordering = "th"
+            #assert self.onnx_attr['pads'] ==
+            #assert [int(i) for i in self.onnx_attr["pads"]] == (0, 0, 0, 0, 0, 0, 0, 0)
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/pad.py]: PadMapper is only a placeholder for now. 
```
for i in self.onnx_attr["pads"]:
     assert int(i) == 0
```

File: pyzoo/zoo/pipeline/api/onnx/mapper/pad.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            pads = (0, 0, 0, 0, 0, 0, 0, 0)
+            padding = (0, 0)
+            pad = zlayers.ZeroPadding2D(padding=padding,
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/pad.py]: `zlayers.ZeroPadding2D(padding=(0, 0)`

File: pyzoo/zoo/pipeline/api/onnx/mapper/pad.py (hunk 4 above)
Comment (zhichao-li) [reply pyzoo/zoo/pipeline/api/onnx/mapper/pad.py]: We might just simply use `Identity` here, but there's no such wrapper at the moment.

File: pyzoo/zoo/pipeline/api/keras/layers/convolutional.py [hunk 6]
```diff
@@ -565,7 +565,7 @@ class ZeroPadding2D(ZooKerasLayer):
     >>> zeropadding2d = ZeroPadding2D(padding=(2, 1), input_shape=(2, 8, 8))
     creating: createZooKerasZeroPadding2D
     """
-    def __init__(self, padding=(1, 1), dim_ordering="th", input_shape=None, **kwargs):
+    def __init__(self, padding, dim_ordering="th", input_shape=None, **kwargs):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/keras/layers/convolutional.py]: why changing this?

File: pyzoo/zoo/pipeline/api/onnx/mapper/pad.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from zoo.pipeline.api.onnx.mapper.operator_mapper import OperatorMapper
+from zoo.pipeline.api.onnx.onnx_helper import OnnxHelper
+import zoo.pipeline.api.keras.layers as zlayers
+
+
+class PadMapper(OperatorMapper):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/pad.py]: do we still need this mapping?

**Comment by xiaxin1998** [comment]
@zhichao-li  Jenkins passed
